lib/astutils.cpp

```diff
@@ -133,9 +133,74 @@ const Token * astIsVariableComparison(const Token *tok, const std::string &comp,
     return ret;
 }
 
+const Token * getVariableExpression(const Variable * var)
+{
+    if(!var || !var->declEndToken())
+        return nullptr;
+    if(Token::Match(var->declEndToken(), "; %varid% =", var->declarationId()))
+        return var->declEndToken()->tokAt(2)->astOperand2();
+    else
+        return var->declEndToken()->astOperand2();
+}
+
+bool isInLoop(const Token * tok) {
+    while(tok->astParent()) {
+        tok = tok->astParent();
+        if(Token::Match(tok->astOperand1(), "for|while"))
```

Comment:
why do you want to look at every operand? How about: `return tok && Token::Match(tok->astTop()->previous(), "for|while (");`